### PR TITLE
Fix centering bug reported in #1322

### PR DESCRIPTION
**Pull request by rmjarvis**
cf. #1322
@beckermr @zhouconghao

### CODE REVIEW

File: galsim/gsobject.py [hunk 1]
```diff
@@ -1225,8 +1225,8 @@ def _get_new_bounds(self, image, nx, ny, bounds, center):
         elif nx is not None and ny is not None:
             b = BoundsI(1,nx,1,ny)
             if center is not None:
-                b = b.shift(_PositionI(np.floor(center.x+0.5)-b.center.x,
-                                       np.floor(center.y+0.5)-b.center.y))
+                b = b.shift(_PositionI(np.floor(center.x+0.5-b.true_center.x),
+                                       np.floor(center.y+0.5-b.true_center.y)))
```
Comment (rmjarvis) [inline galsim/gsobject.py:1229]: Worth mentioning that [here](https://github.com/GalSim-developers/GalSim/blob/releases/2.6/galsim/gsobject.py#L1167) is the other line, which this should be consistent with (and wasn't).

**Review by beckermr — APPROVED** [review]
Thanks for the fix!

I have two related questions, none of them should block this PR but might be good fodder for future work. 

1. Is it worth trying to systematically add tests that exercise all of the drawing options in each combination? I was surprised to find this bug and it reminded me that even if the coverage is near 100%, we can still miss bugs because we have not hit all possible paths through the code. 

2. Is it worth trying to do some internal refactoring on some of the private methods to avoid the need to have certain lines of code be kept in sync? For example, I wonder if it is possible to refactor things such that the wcs, bounds, etc is all made up front and then the image is generated from those. This might help different bits of the code be more robust to changes maybe?

In any case, LGTM and thank you!

**Review by beckermr — CHANGES_REQUESTED** [review]
Ack one other thing. Let's add comments to the two relevant code lines that need to be the same stating that and referring to the other line so future folks understand when reading the code.

**Comment by rmjarvis** [comment]
1. We do try.  We apparently missed these combinations that you discovered.  There were a bunch of other tests that test various combinations of offset and center and such.  But I think mostly with given bounds, rather than creating new ones.  And the ones that tested that combination apparently wasn't comprehensive enough.  If you have further ideas of a test that would be more comprehensive, I'd be happy to add it.
2. Possibly.  I thought about refactoring the `_setup_image` to use `new_bounds` rather than recapitulate the same calculation when necessary.  But (a) I didn't want to gratuitously break any code that uses this private method.  At least not in a bugfix release.  And (b) I wasn't positive that there wasn't a good reason that I didn't already do it that way.

**Comment by beckermr** [comment]
Sounds good. Thank you and LGTM!

**Review by beckermr — APPROVED** [review]

**Pull request merged**